contactsummary/js/src/contact-summary/services/contract.service.js

```diff
@@ -174,6 +175,22 @@ define([
     }
 
     /**
+     * Returns the contract field options
+     *
+     * @param  {string} fieldName
+     * @return {Promise}
+     */
+    function getOptions (fieldName) {
+      var options = settings.CRM.options.HRJobDetails || {};
+
+      if (fieldName && typeof fieldName === 'string') {
+        options = options[fieldName];
```

Comment:
I am not sure what is happening here, what kind data is returned by the API. And why you are selecting a specific option here.

Reply:
Actually the `settings.CRM.options.HRJobDetails` returns the list of field options like following: 
```js
{
  "contract_type": {
    "1": "Fixed Term",
    "2": "Label",
    "Apprentice": "Apprentice",
    "Contractor": "Contractor",
    "Employee - Permanent": "Employee - Permanent",
    "Employee - Temporary": "Employee - Temporary",
    "Intern": "Intern",
    "Trustee": "Trustee",
    "Volunteer": "Volunteer"
  },
  "level_type": false,
  "location": {
    "Headquarters": "Headquarters",
    "Home": "Home or Home-Office"
  },
  "notice_unit": {
    "Day": "Day",
    "Week": "Week",
    "Month": "Month",
    "Year": "Year"
  },
  "notice_unit_employee": {
    "Day": "Day",
    "Week": "Week",
    "Month": "Month",
    "Year": "Year"
  },
  "department": false,
  "end_reason": {
    "1": "Voluntary",
    "2": "Involuntary",
    "3": "Planned",
    "4": "Retirement"
  }
}
```
 So, the giving the user to select the specific options to extract based on `fieldName`. The `fieldName` can be `contract_type` or `end_reason` or `notice_unit_employee` and so on.